### PR TITLE
[pipeline/trusted-resources]Verify task from cluster at taskrun

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit focus on verify the task which is applied in cluster and
referred in taskrun. In this commit we will exclude system populated
fields and keep fields from users. The reason we verify the task in
taskrun validation not the task's validation is that there is a stage
between task creation and taskrun creation, so we need to make sure the
task is not tampered.

Before this commit we can only verify task from oci bundle.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Commit messages includes a project tag in the subject - e.g. [OCI], [hub], [results], [task-loops]

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go [hunk 1]
```diff
@@ -136,8 +131,27 @@ func prepareTask(t v1beta1.TaskObject) (v1beta1.Task, []byte, error) {
...
 		Kind:       "Task"}
 	task.Spec = t.TaskSpec()
-	task.ObjectMeta = t.TaskMetadata()
 
+	// exclude the fields populated by system.
+	task.Name = t.TaskMetadata().Name
+	task.GenerateName = t.TaskMetadata().GenerateName
+	task.Namespace = t.TaskMetadata().Namespace
+
+	task.Labels = make(map[string]string)
```
Comment (wlynch) [inline pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go:140]: In particular, do we need to be sensitive to nil vs empty maps here?

File: pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go [hunk 2]
```diff
@@ -136,8 +131,27 @@ func prepareTask(t v1beta1.TaskObject) (v1beta1.Task, []byte, error) {
 		APIVersion: "tekton.dev/v1beta1",
 		Kind:       "Task"}
 	task.Spec = t.TaskSpec()
-	task.ObjectMeta = t.TaskMetadata()
 
+	// exclude the fields populated by system.
```
Comment (wlynch) [inline pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go:135]: Probably worth adding a test for this func - in particular to make sure we're covering cases like empty/missing data (e.g. metadata, maps, etc.)

**Review by wlynch — CHANGES_REQUESTED** [review]
For the most part looks good! :D

File: pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go (hunk 1 above)
Comment (Yongxuanzhang) [reply pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go:140]: you mean check if t.TaskMetadata().Labels is nil ?

Comment (Yongxuanzhang) [reply pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go:140]: I guess it is unnecessary?  because I don't use labels in my test and examples and they can work 😂  maybe I'm miss something

Comment (Yongxuanzhang) [reply pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go]: I changed it, not sure if that makes sense

**Review by wlynch — APPROVED** [review]
/lgtm
